### PR TITLE
Added 'wait_for' method to get all the images uploaded.

### PR DESCRIPTION
**Pull request by amavinag**
Added 'wait_for' method to get all the images uploaded.

      * wait for all the images to be uploaded on the
        latest directory per stream.
      * if there are no new images are added after wait
        timeout, then it creates the report.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 9a784bbc014e9337c2e6fa4ef986804e611b5b7c:

Everything seems all right :smile: .
*CFME QE Bot*

File: scripts/latest_template_tester_report.py [hunk 1]
```diff
@@ -28,6 +32,14 @@ def parse_cmd_line():
 
 def get_latest_tested_template_on_stream(api, template_stream_name):
     stream = {}
+    try:
+        print "wait until all the provider images are uploaded to latest directory"
+        wait_for(templates_uploaded,
+                 [api, template_stream_name], fail_condition=False, delay=5, timeout='1h')
+    except Exception as e:
+        print e
+        print "less than three provider images are uploaded to latest directory"
```
Comment (mfalesni) [inline scripts/latest_template_tester_report.py]: Please, put parentheses around the print parameters since they are both python2 and python3 compatible.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 84d3a4b103929e6819dcb9a7b52281ad757dff25:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b7ee3e3971fad087fbd5e0074f99ff7ad47f2665:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f6652b64af6a05e3cf802dc557937349d523db91:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 270cd3ce0157e94729dda8820e64d9dff87f0f17:

`scripts/latest_template_tester_report.py`:
- :grey_exclamation: Line 152:47: **S001** *found %s formatter*
- :grey_exclamation: Line 153:52: **S001** *found %s formatter*
- :grey_exclamation: Line 156:47: **S001** *found %s formatter*
- :grey_exclamation: Line 157:52: **S001** *found %s formatter*
- :grey_exclamation: Line 163:47: **S001** *found %s formatter*
- :grey_exclamation: Line 164:52: **S001** *found %s formatter*
- :grey_exclamation: Line 167:47: **S001** *found %s formatter*
- :grey_exclamation: Line 168:52: **S001** *found %s formatter*
- :grey_exclamation: Line 174:47: **S001** *found %s formatter*
- :grey_exclamation: Line 175:52: **S001** *found %s formatter*
- :grey_exclamation: Line 178:47: **S001** *found %s formatter*
- :grey_exclamation: Line 179:52: **S001** *found %s formatter*
- :heavy_exclamation_mark: Line 192:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4e2305a0fc4a98b7ef38313c59756ce17b3c96c2:

`scripts/latest_template_tester_report.py`:
- :bangbang: Line 174:101: **E501** *line too long (102 > 100 characters)*
- :bangbang: Line 178:101: **E501** *line too long (102 > 100 characters)*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a75f243cbe197f3667f6d511c523a6be6c04c1da:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 805401619cc8d78771c379b016441af4155f7131:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**